src/pydap/model.py

```diff
@@ -333,6 +333,9 @@ def keys(self):
         """Method to emulate a dictionary, returning keys."""
         return self._keys[:]
 
+    def __len__(self):
```

Comment:
Consider making `StructureType` inherit from `collections.Mapping` as well. That guarantees you don't forget anything, and helps for type checks (e.g., with `isinstance` or `typing.Mapping`).